### PR TITLE
doc: BEGINNING_DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by nrmancuso**
Creates a "beginning development" document in github, loosely based on https://checkstyle.org/beginning_development.html.  

https://github.com/nrmancuso/checkstyle/blob/beginning-development/docs/BEGINNING_DEVELOPMENT.md

### CODE REVIEW

File: docs/BEGINNING_DEVELOPMENT.md
Comment (nrmancuso) [inline docs/BEGINNING_DEVELOPMENT.md:1]: Note the `docs` directory, this is a github convention that we should start using to have an obvious place where all github docs live; we can consolidate everything in the project root and `.github` directory to this one in a separate issue.

**Comment by nrmancuso** [comment]
Depending on the outcome of https://github.com/checkstyle/checkstyle/issues/14542, we can just delete the website doc in this PR as well.

**Review by romani — APPROVED** [review]
more that a half a year this doc is stuck.
but it supposed to be bring value to who is in need asap.

we can improve doc as we go, as wiki is growing .... update by update , step by step

**Comment by romani** [comment]
There are more room to improve this doc we will do in separate PR.

**Pull request merged**